### PR TITLE
EF, DB connection and more

### PR DESCRIPTION
**Pull request by rawform**
Things added / changed:
+ database is now hosted on https://freeasphosting.net
       MSSQL for local and MySQL for cloud plan didn't work out (very hard to do)
       the choice was MySQL for both (every contributor needs to manually install local MySQL database)
       or MSSQL for both (comes with Visual Studio).
       I chose MSSQL for both.
       Cloud connection string sent privately.
       To connect to cloud -> appsettings.json -> "UseCloudDatabase": true
+ added navigational properties to all models for easier db querying
     Allows usage of .Include() for example. 
* changed GiftCardPaymentId to GiftcardPaymentId, since Giftcard is used everywhere else in the project.
+ added DbContext
+ Entity Framework wants decimal types to have precise decimal places and sends a lot of warnings everytime
       Fixed: all decimal types are now with two decimal places (.00) like most currencies
- DISABLED ALL deletion cascading!!! for archiving purposes

### CODE REVIEW

File: OMGVA PoS/Data layer/Models/Giftcard.cs [hunk 1]
```diff
@@ -1,9 +1,12 @@
 ﻿namespace OMGVA_PoS.Data_layer.Models
 {
-    public class Giftcard
+    public class GiftCard
```
Comment (AugustinasJarockis) [inline OMGVA PoS/Data layer/Models/Giftcard.cs:3]: Why did you change this? It is giftcard in the specification. Also, you did not change the file name.

File: OMGVA PoS/Data layer/Models/Order.cs [hunk 2]
```diff
@@ -9,7 +9,13 @@ public class Order
         public decimal Tip { get; set; }
         public string RefundReason { get; set; }
         public long UserId { get; set; }
-        public long PaymentId { get; set; }
```
Comment (AugustinasJarockis) [inline OMGVA PoS/Data layer/Models/Order.cs:12]: Is there a reason why you specifically deleted this id?

File: OMGVA PoS/Data layer/Models/Payment.cs [hunk 3]
```diff
@@ -4,9 +4,18 @@ namespace OMGVA_PoS.Data_layer.Models
 {
     public class Payment
     {
-        public string Id { get; set; }
+        public long Id { get; set; }
```
Comment (AugustinasJarockis) [inline OMGVA PoS/Data layer/Models/Payment.cs:7]: About this one. In the specification it was mentioned that it is specifically a string. I believe it is because stripe uses strings for ids. If I am correct (I do not know much about stripe), then it should be string.

File: OMGVA PoS/Data layer/Models/Payment.cs [hunk 4]
```diff
@@ -4,9 +4,18 @@ namespace OMGVA_PoS.Data_layer.Models
...
         public PaymentMethod Method { get; set; }
         public long CustomerId { get; set; }
         public long? GiftCardPaymentId { get; set; }
+        public long OrderId { get; set; }
```
Comment (AugustinasJarockis) [inline OMGVA PoS/Data layer/Models/Payment.cs:11]: Why did you add Order Id?

**Review by AugustinasJarockis — CHANGES_REQUESTED** [review]
One more note, the description of the PR seems to be outdated/incorrect.

File: OMGVA PoS/Data layer/Models/Payment.cs (hunk 4 above)
Comment (rawform) [reply OMGVA PoS/Data layer/Models/Payment.cs]: For split payments, removed now

File: OMGVA PoS/Data layer/Models/Payment.cs (hunk 3 above)
Comment (rawform) [reply OMGVA PoS/Data layer/Models/Payment.cs]: Fair enough, reverted, changed PaymentId in Order.cs to string as well

File: OMGVA PoS/Data layer/Models/Order.cs (hunk 2 above)
Comment (rawform) [reply OMGVA PoS/Data layer/Models/Order.cs:12]: For split payments, reverted now

File: OMGVA PoS/Data layer/Models/Giftcard.cs (hunk 1 above)
Comment (rawform) [reply OMGVA PoS/Data layer/Models/Giftcard.cs]: Since gift card is two words it should be GiftCard. Reverted now, however, done according to specification

**Review by GitaGliaudyte — APPROVED** [review]
